### PR TITLE
Some cwl changes

### PR DESCRIPTION
**Pull request by muzimuzhi**
__Main changes__

Full updates, means the whole cwl file is updated according to the doc and source of corresponding latest package:
 - update etoolbox.cwl to v2.5e
 - update menukeys.cwl to v1.5
 - update fontspec.cwl to v2.7c
 - update booktabs to v1.618033
 - update amsthm to v2.20.3

Partial changes:
 - add starred variants for `\providecommand`, `\newenvironment` and `\renewenvironment`
 - remove duplicate lines of `\newtheorem`, which is defined in latex2e and extended to starred version by `amsthm`

Style changes:
 - to reduce date ambiguity, use date format `d mmm yyyy` (e.g., `1 Sep 2019`) consistently

### CODE REVIEW

File: completion/booktabs.cwl [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
+# package booktabs
 # dani/2006-02-18
-# muzimuzhi/2019-07-17
-\addlinespace
-\addlinespace[space]
+# muzimuzhi/17 Jul 2019
+# muzimuzhi/1 Sep 2019 for booktabs v1.618033
+
+\toprule
+\toprule[line width]
```
Comment (sunderme) [inline completion/booktabs.cwl]: an argument 'width' is recognized as latex-length, 'line width' is not.
So, please either revert to 'width' or 'line width%l'

File: completion/booktabs.cwl [hunk 2]
```diff
@@ -1,16 +1,20 @@
...
-\midrule[width]
+\cmidrule[line width](trim){a-b}
+
 \morecmidrules
-\specialrule{width}{abovespace}{belowspace}
-\toprule
-\toprule[width]
+\specialrule{line width}{above space}{below space}
+\addlinespace
+\addlinespace[vspace]
```
Comment (sunderme) [inline completion/booktabs.cwl]: also a width which is derived from 'width','length' or 'height'

**Comment by muzimuzhi** [comment]
Changed.

Maybe a more clear documentation about cwl notation format is needed. I can try it.

**Comment by sunderme** [comment]
thanks.
The user manual , chapter 4.13.2 describes the arguments. Not sure how that needs to be improved.

**Comment by muzimuzhi** [comment]
When preparing the changes about cwl fils, I read the sec. 4.13 of user manual very often, and think if we can re-arrange the info to make it clearer and easier to use. I will open new issue If I came up with some possible solutions.

**Pull request merged**